eZ/Publish/Core/Limitation/SiteAccessLimitationType.php

```diff
@@ -82,7 +81,7 @@ public function validate(APILimitationValue $limitationValue)
     {
         $validationErrors = [];
         foreach ($limitationValue->limitationValues as $key => $value) {
-            if (!isset($this->siteAccessList[$value])) {
+            if (!isset($this->getSiteAccessList()[$value])) {
```

Comment:
Please decrease its complexity back to `O(n)` by moving `$this->getSiteAccessList()` outside of the loop.

Reply:
`\eZ\Publish\Core\MVC\Symfony\SiteAccess\SiteAccessServiceInterface::exists` could be used instead. 

> Please decrease its complexity back to O(n) by moving $this->getSiteAccessList() outside of the loop.

What kind of complexity do you have in mind @alongosz ? :smirk: